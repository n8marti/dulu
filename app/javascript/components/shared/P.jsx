import React from "react";
import PropTypes from "prop-types";
import style from "./P.css";

export default function P(props) {
  return <div className={style.P}>{props.children}</div>;
}

P.propTypes = {
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.object])
};
