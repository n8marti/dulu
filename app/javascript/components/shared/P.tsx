import React from "react";
import style from "./P.css";
import { Children } from "../../models/TypeBucket";

interface IProps {
  children: Children;
}

export default function P(props: IProps) {
  return <div className={style.P}>{props.children}</div>;
}
