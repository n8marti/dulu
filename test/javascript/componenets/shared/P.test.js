import React from "react";
// import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import P from "components/shared/P";

test("rendered componenet", () => {
  const wrapper = shallow(
    <P>
      <div>Here are some words.</div>
    </P>
  );
  expect(wrapper).toMatchSnapshot();
});
