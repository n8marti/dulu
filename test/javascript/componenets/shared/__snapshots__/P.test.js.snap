// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered componenet 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <P>
    <div>
      Here are some words.
    </div>
  </P>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        Here are some words.
      </div>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Here are some words.",
      },
      "ref": null,
      "rendered": "Here are some words.",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          Here are some words.
        </div>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Here are some words.",
        },
        "ref": null,
        "rendered": "Here are some words.",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
